common/djangoapps/student/models.py

```diff
@@ -1621,6 +1668,12 @@ def user_can_skip_entrance_exam(cls, user, course_key):
 class LanguageProficiency(models.Model):
     """
     Represents a user's language proficiency.
+
+    Note that we have not found a way to emit analytics change events by using signals directly on this
+    model or on UserProfile. Therefore if you are changing LanguageProficiency values, it is important
```

Comment:
No, this comment is correct. We haven't been able to emit changes about LanguageProficiency (which is a 1 to many table) through signal handlers. We tried doing it both by signal handlers on LanguageProficiency, and on UserProfile, which has a ForeignKey relationship.

Happy to provide gory details if interested. In summary, we want to event the complete set of proficiencies for a user, not individual adds and deletes.